src/realm/realmReducer.js

```diff
@@ -177,6 +183,22 @@ export default (
           //   `op: 'update_dict'` events near the edge.)
           return state;
 
+        case EventTypes.user_settings:
+          if (event.op === 'update') {
+            const { property, value } = event;
+            switch (property) {
+              case 'twenty_four_hour_time': {
+                invariant(typeof value === 'boolean', 'expected boolean for twenty_four_hour_time');
+                return { ...state, twentyFourHourTime: value };
```

Comment:
Hmm, looking again, yeah: if the `invariant` ever failed, it might not mean a bug in the zulip-mobile code. In fact it most likely wouldn't; it'd most likely mean a server was behaving unexpectedly. There isn't a lot we do to this value between the edge and here, and I think that's basically thanks to following the pattern of 979d28358 in `eventToAction`.

When I wrote this, I must've been thinking something like, "Hmm, I've failed to tell Flow quite what my expectations are; I haven't yet managed to write the type I actually intend. (In this case, `property` and `value` should correspond but they don't.) But surely my expectations are correct, and we have a TODO on the type to write it the way we mean. So I'll just use an `invariant` to satisfy Flow until we fix that."

When I run through that thought process, it might be the case (as it turned out to be this time) that "my expectations" follow from expectations about how servers behave. I think I agree that we shouldn't use `invariant` in these cases, for the reasons you mention. I do have to remember to check if I'm in one of those cases, and to check attentively. 🙂

> Specifically, let's log a warning and then ignore the event. We do this in a few similar places already; e.g. `src/notification/notifOpen.js` on an invalid URL for `realm_uri`.

What do you think about a cast and `$FlowFixMe` instead:

```js
  // $FlowFixMe[incompatible-cast] - fix UserSettingsUpdateEvent
  return { ...state, twentyFourHourTime: (value: boolean) };
```

You've pointed out another place in this PR where I've done basically that, without logging a warning. I think in this case, as in that one, it's unlikely that servers will defy our expectations. Certainly that isn't made more likely by my `UserSettingsUpdateEvent` being slightly wrong. 😛

Reply:
Yeah, that's fine.